lib/tlTimeline/IRender.h

```diff
@@ -48,6 +48,9 @@ namespace tl
             //! Finish a render.
             virtual void end() = 0;
 
+            //! Sets the mesh matrix to a new value
+            virtual void setMatrix(const math::Matrix4x4f&) = 0;
```

Comment:
I don't think having "setMatrix()" is enough, there is no way of returning it to the previous state. Maybe what we need is "pushMatrix()" and "popMatrix()" so the matrix can be modified and then reverted. The matrix should also apply to all of the primitives for consistency, or there should at least be a note in the comments about the limitations.

Reply:
I was unsure what shaders are used only in the display pipeline and what shaders might be used both in the FBO texture creation as well as in the viewport rendering/baking.  I thought only the "mesh" shader was used.

I think that the pushMatrix/popMatrix functions is overkill.  You'd be emulating the opengl2 functionality.

Reply:
Probably all of the drawing functions could be rewritten using the mesh functionality, that would reduce the complexity and provide a common code path.

Adding "pushMatrix()" and "popMatrix()" may be overkill, but I don't think the "setMatrix()" function is sufficient for the reasons I gave above.

Reply:
Okay.  I'll add a pushMatrix() and popMatrix() functionality.  That means I need to create Shader::getUniform() functions first, right?